src/rdkafka_queue.c

```diff
@@ -118,6 +124,33 @@ rd_kafka_q_t *rd_kafka_q_new0(rd_kafka_t *rk, const char *func, int line) {
         return rkq;
 }
 
+/*
+ * Handles the logic for increment/decrement of rkq_consumer_cnt.
+ * Only used internally when forwarding queues.
+ * @locks rd_kafka_q_lock(rkq)
+ */
+static void rd_kafka_q_consumer_cnt_propagate(rd_kafka_q_t *rkq,
+                                              int cnt_delta) {
+        rd_assert(rkq);
+        if (cnt_delta == 0)
+                return;
+
+        mtx_lock(&rkq->rkq_lock);
+        rkq->rkq_consumer_cnt += cnt_delta;
+        rd_assert(rkq->rkq_consumer_cnt >= 0);
```

Comment:
Use warn log instead of assert

Reply:
Remove this statement entirely because we're now using the flag